node_package/src/clientStartup.ts

```diff
@@ -233,7 +233,12 @@ function unmount(el: Element): void {
 function reactOnRailsPageUnloaded(): void {
   debugTurbolinks('reactOnRailsPageUnloaded');
   if (supportsRootApi) {
-    for (const root of findContext().roots) {
+    const roots = findContext().roots;
+
+    // If no react on rails components
+    if (!roots) return;
+
+    for (const root of roots) {
```

Comment:
Looks OK for me. @alexeyr-ci1 @alexeyr please review as you added the prior line for React 18.

This seems harmless, so I'll merge. Still, I'd like a post-merge review.